libsweep/src/sweep.cc

```diff
@@ -15,6 +67,8 @@ typedef struct sweep_error {
 typedef struct sweep_device {
   sweep::serial::device_s serial; // serial port communication
   bool is_scanning;
+  ScanQueue* scan_queue;
```

Comment:
Why a raw pointer? Who's responsible for ownership here?

Either make ScanQueue default constructible or use a unique_ptr